migration/idempotent/010-telemetry.sql

```diff
@@ -116,6 +116,12 @@ $$
         SELECT public.approximate_row_count('_ps_trace.span')::TEXT INTO result;
         PERFORM _ps_catalog.apply_telemetry('traces_spans_total_approx', result);
 
+        -- Trace spans processed by jaeger collector will always have an internal attribute named 'internal.span.format' with possible values 'jaeger|zipkin|proto' according to https://github.com/jaegertracing/jaeger/issues/1490. We can use this to infer whether promscale's gRPC remote storage implementation has been used or not.
+        SELECT ARRAY_AGG(value)::TEXT INTO result FROM _ps_trace.tag WHERE key='internal.span.format';
```

Comment:
I have two problems with the proposed query:
- Firstly, I don't like that it puts an unordered array of potentially user-supplied values into the metadata table. E.g. nothing prevents a value from being a megabyte of garbage, even though it's not a common case. It's also a way for us to accidentally leak customer's data.
- Secondly, bitmap scan is suboptimal, since we know the values you expect. It should be possible to put both a key and a `_prom_ext.jsonb_digest` of value into the `WHERE` clause and get a plan with an index-only scan.

Reply:
I think the metric per type approach discussed above may work if you further restrict it to a set of expected types.

Reply:
A boolean would also work

Reply:
@sumerman Nice catch.

How about `SELECT ARRAY_AGG(value)::TEXT FROM _ps_trace.tag WHERE key='internal.span.format' and value in ('"jaeger"', '"zipkin"','"proto"');` ?

It actually uses index too,

```
Aggregate  (cost=3.58..3.60 rows=1 width=32) (actual time=0.063..0.064 rows=1 loops=1)
  ->  Index Scan using tag_key_value_id_key_id_key_idx on tag  (cost=0.27..3.58 rows=1 width=25) (actual time=0.026..0.032 rows=2 loops=1)
        Index Cond: ((key)::text = 'internal.span.format'::text)
        Filter: (value = ANY ('{"\"jaeger\"","\"zipkin\"","\"proto\""}'::jsonb[]))
Planning Time: 0.482 ms
Execution Time: 0.123 ms
```